<?xml version="1.0"?>
<launch>

  <arg name="pacmod_param_path" default="$(find-pkg-share pacmod_interface)/config/pacmod.param.yaml"/>
  <arg name="pacmod_extra_param_path" default="$(find-pkg-share pacmod_interface)/config/pacmod_extra.param.yaml"/>


  <!-- pacmod interface -->
  <node pkg="pacmod_interface" exec="pacmod_interface" name="pacmod_interface" output="screen">
    <param from="$(var pacmod_param_path)" />
  </node>

  <!-- pacmod diagnostics publisher -->
  <node pkg="pacmod_interface" exec="pacmod_diag_publisher" name="pacmod_diag_publisher" output="screen">
  </node>

  <!-- pacmod additional parameter changer -->
  <node pkg="pacmod_interface" exec="pacmod_dynamic_parameter_changer" name="pacmod_dynamic_parameter_changer" output="screen">
    <param from="$(var pacmod_extra_param_path)" />
    <remap from="~/output/can" to="/pacmod/can_rx"/>
    <remap from="~/input/steer_rpt" to="/pacmod/steer_rpt"/>
  </node>

</launch>
